### PR TITLE
IMPRO-1446: removing extrapolate from optical flow

### PR DESCRIPTION
**Pull request by neilCrosswaite**
# Removed extrapolate from optical flow
- only calls optical-flow
- stops the saving of the extrapolated files
- remove tests
- removed argparser args

# Saved the 2 files as a cubelist rather than 2 cubes
- process returns cubelist
- saves that cubelist
- update tests
- make argparser take n output argument for the output filepath.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/cli/nowcast_optical_flow.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4484394&bid=15129354)

File: lib/improver/cli/nowcast_optical_flow.py [hunk 1]
```diff
@@ -88,57 +86,25 @@ def main(argv=None):
...
-                    args.output_dir, generate_file_name(cube))
-            save_netcdf(cube, file_name)
+    save_netcdf(result, args.output_filepath)
 
 
 def process(original_cube_list, orographic_enhancement_cube=None,
             attributes_dict=None, ofc_box_size=30,
-            smart_smoothing_iterations=100, extrapolate=False,
-            max_lead_time=360, lead_time_interval=15):
+            smart_smoothing_iterations=100):
```
Comment (gavinevans) [inline lib/improver/cli/nowcast_optical_flow.py]: The docstring for this method needs to be updated to remove the references to extrapolation, as this function is only computing the optical flow now.

File: lib/improver/cli/nowcast_optical_flow.py [hunk 2]
```diff
@@ -61,10 +60,9 @@ def main(argv=None):
                         "T-2 from which to calculate optical flow velocities. "
                         "The files require a 'time' coordinate on which they "
                         "are sorted, so the order of inputs does not matter.")
-    parser.add_argument("--output_dir", metavar="OUTPUT_DIR", type=str,
-                        default='', help="Directory to write all output files,"
-                        " or only advection velocity components if "
-                        "NOWCAST_FILEPATHS is specified.")
+    parser.add_argument("output_filepath", metavar="OUTPUT_FILEPATH",
```
Comment (gavinevans) [inline lib/improver/cli/nowcast_optical_flow.py]: The `Argparser` description needs to be updated to remove the "and (optionally) extrapolate to required lead times".

Comment (gavinevans) [inline lib/improver/cli/nowcast_optical_flow.py]: The docstring for the `main` function needs to be updated to remove " and (optionally) extrapolate data".

Comment (neilCrosswaite) [reply lib/improver/cli/nowcast_optical_flow.py]: Done

Comment (neilCrosswaite) [reply lib/improver/cli/nowcast_optical_flow.py]: Done

File: lib/improver/cli/nowcast_optical_flow.py (hunk 1 above)
Comment (neilCrosswaite) [reply lib/improver/cli/nowcast_optical_flow.py]: Done.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/cli/nowcast_optical_flow.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4484394&bid=15129354)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/cli/nowcast_optical_flow.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4484394&bid=15129354)

**Pull request merged**